src/fe_modes.c

```diff
@@ -122,6 +122,25 @@ void switch_mode(Mode_ID new_mode, State *state) {
   refresh_screen();
 }
 
+Mode_ID add_mod_canvas_mode(Mode_ID mode, int n) {
+  int mode_first = MODE_PICKER + 1;  // beginning of selectable modes
+  int mode_list_end = LAST;          // length of total mode list
+
+  if (mode < mode_first || mode_list_end <= mode) {
+    logd("mode \"%s\" is not a canvas mode", mode);
+    return mode;
+  }
+
+  // loop properly between mode_first and last mode
+  return ((mode - mode_first) + n) % (mode_list_end - mode_first) + mode_first;
+}
+
+Mode_ID next_canvas_mode(Mode_ID mode) { return add_mod_canvas_mode(mode, 1); }
```

Comment:
Linter makes this look dumb with the one below, but not worth changing?

Reply:
I looked into this a little more, the option is `AllowShortFunctionsOnASingleLine`. I'll make another pull request to change it.